File: python/candyland/deck.py
from random import shuffle


class Deck(object):
	
	def __init__(self):
		#super(Deck, self).__init__()
		#self.arg = arg
		self.deck = []
		self.discard = []

		self.build_deck()
		self.shuffle_deck()

	def build_deck(self):
		
		#do 6 times 1 2 3 4 5 6
		for i in range (1,6):	
			colors = ["Red","Purple","Yellow","Blue","Orange","Green"]
			for color in colors:
				self.deck.append(Card(color,'',1))
		
		for i in range (1,4):
			colors = ["Red","Purple","Yellow","Blue"]
			for color in colors:
				self.deck.append(Card(color,'',2))
		for i in range (1,3):
			colors =["Orange","Green"]
			for color in colors:
				self.deck.append(Card(color,'',2))

		self.deck.append(Card('Pink','Plumpy',1))
		self.deck.append(Card('Pink','Mr Mint',1))
		self.deck.append(Card('Pink','Jolly',1))
		self.deck.append(Card('Pink','Gramma Nutt',1))
		self.deck.append(Card('Pink','Princess Lolly',1))
		self.deck.append(Card('Pink','Queen Frostine',1))

	
	def shuffle_deck(self):	
		shuffle(self.deck)
		shuffle(self.deck)
		shuffle(self.deck)
		shuffle(self.deck)

	def reshuffle(self):
		self.deck = self.discard
		self.discard = []
		self.shuffle_deck()

	def draw_card(self):
		card_count = len(self.deck)
		if (card_count == 0):
			self.reshuffle()
		drawn_card = self.deck.pop()
		self.discard.append(drawn_card)
		return drawn_card

class Card(object):
	def __init__(self, color,label,move_count):
		#super(Card, self).__init__()
		self.color = color
		self.label = label
		self.move_count = move_count

	def to_string(self):
		if self.move_count == 1 and self.color != "Pink":
			return "Single %s" % self.color
		elif self.move_count == 2:
			return "Double %s" % self.color
		else:
			return "%s" % self.label

File: java/candyland/Deck.java
package candyland;
import candyland.Card;
import java.util.*;

public class Deck {
	List<Card> deck;
	List<Card> discard;

	public Deck() {
		deck = new ArrayList<Card>();
		discard = new ArrayList<Card>();

		build_deck();
		shuffle_deck();
	}

	public Card draw_card() {
		if (this.deck.size() == 0) {
			System.out.println("Reshuffling Deck");
			reshuffle_deck();
		}
		Card drawn_card = this.deck.remove(0);
		this.discard.add(drawn_card);
		return drawn_card;

	}

	private void build_deck() {
		for (int loopCount = 0; loopCount < 6;loopCount++) {
			String[] colors = {"Red","Purple","Yellow","Blue","Orange","Green"};
			for (String color : colors) {
				//add single colors
				this.deck.add(new Card(color,null,1));
			}
		}
		for (int loopCount = 0; loopCount < 4;loopCount++) {
			String[] colors = {"Red","Purple","Yellow","Blue"};
			for (String color : colors ) {
				// add double colors
				this.deck.add(new Card(color,null,2));
			}
		}
		for (int loopCount = 0; loopCount < 3;loopCount++) {
			String[] colors = {"Orange","Green"};
			for (String color : colors) {
				// add the rest of the double colors
				this.deck.add(new Card(color,null,2));
			}
		}

		// add character cards
		this.deck.add(new Card("Pink","Plumpy",1));
		this.deck.add(new Card("Pink","Mr Mint",1));
		this.deck.add(new Card("Pink","Jolly",1));
		this.deck.add(new Card("Pink","Gramma Nutt",1));
		this.deck.add(new Card("Pink","Princess Lolly",1));
		this.deck.add(new Card("Pink","Queen Frostine",1));

	}

	private void shuffle_deck() {
		Collections.shuffle(this.deck);
		
	}

	private void reshuffle_deck() {
		this.deck = new ArrayList<Card>(this.discard);
		this.discard.clear();
		shuffle_deck();

	}
}